### PR TITLE
Minor changes and fixes

### PR DESCRIPTION
**Pull request by alexdewar**
Hi Jamie,

The main point of this PR is to fix a small bug I found in the ``Tank::drive()`` function: the value of ``r`` should be divided by ``sqrt(2)``, otherwise we get out-of-range motor values and now with the new ``BOB_ASSERT()``s this breaks things :-)

I've also included some plotting code and a fix for the problem where ``KEY_UP`` and ``KEY_DOWN`` macros were breaking OpenCV as well as some other small bits.

### CODE REVIEW

**Comment by alexdewar** [comment]
I've decided it makes more sense to just cap the value of ``r`` at one. Otherwise when you push the joystick straight up, the motors will only go at ``1/sqrt(2)`` of maximum speed.

File: common/plot_agent.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// Third-party includes
+#include "../third_party/matplotlibcpp.h"
+#include "../third_party/units.h"
+
+// Standard C++ includes
+#include <array>
+
+namespace BoBRobotics {
+    template<typename AgentType>
+    void plotAgent(const AgentType &agent, std::array<double, 2> &&xlim, std::array<double, 2> &&ylim)
```
Comment (neworderofjamie) [inline common/plot_agent.h]: How come you're passing these as an rvalue reference? isn't this just a case for ``const std::array<double, 2> &xlim``?

File: hid/joystick_linux.h [hunk 2]
```diff
@@ -1,24 +1,35 @@
...
+/*
+ * We need to undef these macros (defined in linux/joystick.h) because they
+ * break OpenCV.
+ */
+#ifdef KEY_UP
+#undef KEY_UP
+#endif
+#ifdef KEY_DOWN
+#undef KEY_DOWN
+#endif
```
Comment (neworderofjamie) [inline hid/joystick_linux.h]: Nice work figuring out where this was coming from!

File: common/plot_agent.h (hunk 1 above)
Comment (alexdewar) [reply common/plot_agent.h]: Hmm yeah it probably should be :-/

**Pull request merged**